clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -1351,10 +1876,20 @@ Expected<SmallVector<OffloadFile>> getDeviceInput(const ArgList &Args) {
         sys::fs::is_directory(*Filename))
       continue;
 
+    // Some of the object files may be bundled using clang-offload-bundler
+    // Following code tries to unbundle these files.
+    auto UnbundledFile = sycl::unbundle(*Filename, Args);
+    if (!UnbundledFile)
+      return UnbundledFile.takeError();
+    auto IRFile = (*UnbundledFile == *Filename)
+                      ? *Filename
+                      : sycl::convertSPIRVToIR(*UnbundledFile, Args);
```

Comment:
To me, a comment here would useful to explain why exactly we expect SPIR-V in files bundled by `clang-offload-bundler`